pkg/backends/locust/spec.go

```diff
@@ -15,13 +17,21 @@ var (
 )
 
 //BuildLoadTestSpec validates input and returns valid LoadTestSpec
-func BuildLoadTestSpec(overwrite bool, distributedPods int32, tags loadTestV1.LoadTestTags, testFileStr, envVarsStr, targetURL string, duration time.Duration) (loadTestV1.LoadTestSpec, error) {
+func BuildLoadTestSpec(overwrite bool, distributedPods int32, tags loadTestV1.LoadTestTags, testFileStr, envVarsStr, targetURL string, duration time.Duration, masterImageRef reference.NamedTagged) (loadTestV1.LoadTestSpec, error) {
 	lt := loadTestV1.LoadTestSpec{}
 	if distributedPods <= int32(0) {
 		return lt, ErrRequireMinOneDistributedPod
 	}
 	if testFileStr == "" {
 		return lt, ErrRequireTestFile
 	}
-	return loadTestV1.NewSpec(loadTestV1.LoadTestTypeLocust, overwrite, distributedPods, tags, testFileStr, "", envVarsStr, loadTestV1.ImageDetails{}, loadTestV1.ImageDetails{}, targetURL, duration), nil
+	// Use defaults if unspecified
+	masterImage := loadTestV1.ImageDetails{Image: defaultImage, Tag: defaultImageTag}
+	if masterImageRef != nil {
+		masterImage = loadTestV1.ImageDetails{
+			Image: masterImageRef.Name(),
+			Tag:   masterImageRef.Tag(),
+		}
+	}
+	return loadTestV1.NewSpec(loadTestV1.LoadTestTypeLocust, overwrite, distributedPods, tags, testFileStr, "", envVarsStr, masterImage, loadTestV1.ImageDetails{}, targetURL, duration), nil
```

Comment:
I think Locust uses same docker image for master-worker, @diegomarangoni can you confirm?

Reply:
Yes, they both use same image

Reply:
Resolved via cfc47ab.